node-src/git/getCommitAndBranch.ts

```diff
@@ -157,6 +160,16 @@ export default async function getCommitAndBranch(
     branch = branch.replace(ORIGIN_PREFIX_REGEXP, '');
   }
 
+  const mergeQueueBranchPrNumber = await mergeQueueBranchMatch(commit.commit, branch)
+
+  if (mergeQueueBranchPrNumber) {
+    const branchFromMergeQueuePullRequestNumber = await getBranchFromMergeQueuePullRequestNumber(ctx, { number: mergeQueueBranchPrNumber });
+   
+    if (branchFromMergeQueuePullRequestNumber) {
+      branch = branchFromMergeQueuePullRequestNumber;
+    }
+  }
```

Comment:
The only thing I would suggest is maybe adding a comment here to explain what's going on.